### DISCUSSION TITLE
How to handle background process

### DISCUSSION THREAD

**Discussion by SomaxSoftware** [JonPSmith/AuthPermissions.AspNetCore#61] [Q&A] [answered]
Hello John,
many thanks for this great library!
We implemented a singlelevel tenant application.
For background processes we need the ability to access the data of the tenants.
What is your opinion, what would be the best approch to link in background to a tenant?

Many thanks for your opinion!
Boris

**Comment by JonPSmith** [comment, marked as answer]
As you know background processes don't have a valid user for obvious reasons. I would suggest you create the DbContext manually in your background process. You would need to obtain the `DbContextOptions<YourTenantDbContext>` service via DI and then use the code below to create the `YourTenantDbContext`.

```c#
using var context = new YourTenantDbContext(optionsViaDi, "The correct DataKey for the tenant you want to access")
```

**Reply by SomaxSoftware** [reply]
Ok, sounds very interesting...
Many thanks! :-)
